contracts/libraries/Positions.sol

```diff
@@ -235,9 +234,6 @@ library Positions {
                 true
             );
             if (params.fungible && params.amount > 0) {
-                
-                amount0Removed = uint128(uint256(amount0Removed) * uint256(params.amount) / (cache.totalSupply + params.amount));
-                amount1Removed = uint128(uint256(amount1Removed) * uint256(params.amount) / (cache.totalSupply + params.amount));
                 params.collect = true;
```

Comment:
Here we were taking a partial amount of a partial amount.

We already calculated the liquidityAmount reflective of the token balance so there is no need to do anything further.